.github/workflows/ci.yml

```diff
@@ -12,6 +12,19 @@ on:
   workflow_dispatch:
 
 jobs:
+
   ci:
-    uses: "puppetlabs/cat-github-actions/.github/workflows/gem_ci.yml@main"
+    uses: "puppetlabs/cat-github-actions/.github/workflows/gem_ci.yml@maint-generic_gem_task"
```

Comment:
what is the reason for referencing a specific branch? Is this one of the benefits of cat-github-actions? That we can use different versions of the same workflow by pointing to different branches?

Reply:
It is indeed.. however this PR is not ready for review yet - it should be draft but I need the jobs to trigger (easily).